managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -3633,12 +3633,19 @@ public boolean isValidPosition(PositionImpl position) {
             log.debug("IsValid position: {} -- last: {}", position, last);
         }
 
-        if (position.getEntryId() < 0) {
+        if (!ledgers.containsKey(position.getLedgerId())){
```

Comment:
Do we need to add a test for covering this branch?

Reply:
Sure

Reply:
Added